crates/rpc-types-engine/src/payload/mod.rs

```diff
@@ -19,3 +22,72 @@ pub enum OpExecutionPayload {
     /// V4 payload
     V4(OpExecutionPayloadV4),
 }
+
+impl OpExecutionPayload {
+    /// Returns a reference to the V2 payload, if any.
+    pub const fn as_v2(&self) -> &ExecutionPayloadV2 {
+        match self {
+            Self::V2(payload) => payload,
+            Self::V3(payload) => &payload.payload_inner,
+            Self::V4(payload) => todo!(),
+        }
+    }
+
+    /// Returns a mutable reference to the V2 payload, if any.
+    pub fn as_v2_mut(&mut self) -> &ExecutionPayloadV2 {
+        match self {
+            Self::V2(payload) => payload,
+            Self::V3(payload) => &mut payload.payload_inner,
+            Self::V4(payload) => todo!(),
+        }
+    }
+    /// Returns the parent hash for the payload.
+    pub const fn parent_hash(&self) -> B256 {
+        self.as_v2().payload_inner.parent_hash
+    }
+
+    /// Returns the block hash for the payload.
+    pub const fn block_hash(&self) -> B256 {
+        self.as_v2().payload_inner.block_hash
+    }
+
+    /// Returns the block number for this payload.
+    pub const fn block_number(&self) -> u64 {
+        self.as_v2().payload_inner.block_number
+    }
+
+    /// Converts [`OpExecutionPayload`] to [`Block`].
+    ///
+    /// Caution: This does not set fields that are not part of the payload and only part of the
+    /// [`OpExecutionPayloadSidecar`]:
+    /// - parent_beacon_block_root
+    ///
+    /// See also: [`OpExecutionPayload::try_into_block_with_sidecar`]
+    pub fn try_into_block(self) -> Result<OpBlock, PayloadError> {
+        match self {
+            Self::V2(payload) => payload.try_into_block(),
+            Self::V3(payload) => payload.try_into_block(),
+            Self::V4(payload) => payload.payload_inner.try_into_block(),
```

Comment:
when https://github.com/alloy-rs/op-alloy/pull/435 is merged we need
```suggestion
            Self::V4(payload) => payload.try_into_block(),
```